sql-plugin/src/main/spark341db/scala/com/nvidia/spark/rapids/shims/Spark341PlusDBShims.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "341db"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.shims
+
+import com.nvidia.spark.rapids._
+import com.nvidia.spark.rapids.GpuOverrides.pluginSupportedOrderableSig
+
+import org.apache.spark.rapids.shims.GpuShuffleExchangeExec
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.plans.physical.SinglePartition
+import org.apache.spark.sql.execution.{CollectLimitExec, GlobalLimitExec, SparkPlan, TakeOrderedAndProjectExec}
+import org.apache.spark.sql.execution.exchange.ENSURE_REQUIREMENTS
+import org.apache.spark.sql.rapids.GpuV1WriteUtils.GpuEmpty2Null
+
+trait Spark341PlusDBShims extends Spark332PlusDBShims {
+
+  override def getExprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = {
+    val shimExprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = Seq(
+      // Empty2Null is pulled out of FileFormatWriter by default since Spark 3.4.0,
+      // so it is visible in the overriding stage.
+      GpuOverrides.expr[Empty2Null](
+        "Converts the empty string to null for writing data",
+        ExprChecks.unaryProjectInputMatchesOutput(
+          TypeSig.STRING, TypeSig.STRING),
+        (a, conf, p, r) => new UnaryExprMeta[Empty2Null](a, conf, p, r) {
+          override def convertToGpu(child: Expression): GpuExpression = GpuEmpty2Null(child)
+        }
+      )
+    ).map(r => (r.getClassFor.asSubclass(classOf[Expression]), r)).toMap
+    super.getExprs ++ shimExprs ++ DayTimeIntervalShims.exprs ++ RoundingShims.exprs
+  }
+
```

Comment:
This code is the same in the 2 places extending Spark332PlusDBShims, let us move it there and delete from Spark341PlusDBShims and sql-plugin/src/main/spark332db/scala/com/nvidia/spark/rapids/shims/SparkShims.scala

Reply:
Can't do that as the Empty2Null is in a different package.